From 92f56777b3ebbf780ef55f64e23f84a34ce71f21 Mon Sep 17 00:00:00 2001
From: Erik Faye-Lund <erik.faye-lund@collabora.com>
Date: Fri, 31 Oct 2025 09:52:36 +0100
Subject: [PATCH] mesa/st: do not enable EXT_texture_buffer_object with rgba
 only
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

GL_EXT_texture_buffer_object requires support for alpha, luminance,
luminance-alpha and intensity formats. If we can't support those, we
can't enable the extension.

Fixes: 45ca7798dc3 ("glsl: handle interactions between EXT_gpu_shader4 and texture extensions")
Reviewed-by: Lars-Ivar Hesselberg Simonsen <lars-ivar.simonsen@arm.com>
Reviewed-by: Iago Toral Quiroga <itoral@igalia.com>
Reviewed-by: Marek Olšák <marek.olsak@amd.com>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/38162>
(cherry picked from commit 6f2b8c3f615af6a38b3da3fb5094b8b4390e64f2)
---
 .pick_status.json                      | 2 +-
 src/mesa/state_tracker/st_extensions.c | 4 +++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/.pick_status.json b/.pick_status.json
index 113ac6ca46683..5910932fc1ea7 100644
--- a/.pick_status.json
+++ b/.pick_status.json
@@ -334,7 +334,7 @@
         "description": "mesa/st: do not enable EXT_texture_buffer_object with rgba only",
         "nominated": true,
         "nomination_type": 2,
-        "resolution": 0,
+        "resolution": 1,
         "main_sha": null,
         "because_sha": "45ca7798dc32c1cb7da8f94af9a7d7400ee9bc12",
         "notes": null
diff --git a/src/mesa/state_tracker/st_extensions.c b/src/mesa/state_tracker/st_extensions.c
index f5f360eb80c67..15a1c913181fa 100644
--- a/src/mesa/state_tracker/st_extensions.c
+++ b/src/mesa/state_tracker/st_extensions.c
@@ -1265,7 +1265,9 @@ void st_init_extensions(struct pipe_screen *screen,
        * pipe cap.
        */
       extensions->EXT_gpu_shader4 = GL_TRUE;
-      extensions->EXT_texture_buffer_object = GL_TRUE;
+
+      if (!screen->caps.buffer_sampler_view_rgba_only)
+         extensions->EXT_texture_buffer_object = GL_TRUE;
 
       if (consts->MaxTransformFeedbackBuffers &&
           screen->caps.shader_array_components)
-- 
GitLab

